pkg/util/kubeobjects/internal/query/query.go

```diff
@@ -63,6 +69,11 @@ func (c Generic[T, L]) Update(ctx context.Context, object T) error {
 		}
 	}
 
+	err := hasher.AddAnnotation(object)
+	if err != nil {
+		return errors.WithStack(err)
+	}
```

Comment:
some hints:

in the case you call `CreateOrUpdate`, will you ever reach this?
- you will not
- that is why the tests you updated failed

why?
- does the "newly generated" object has a hash, that will be used check equality in `CreateOrUpdate`?

would it be safe to use `hasher.AddAnnotation(object)` multiple times in a row?
- yes, if done correctly

Reply:
https://github.com/Dynatrace/dynatrace-operator/pull/3983/commits/5ec6fddaef04bf927b108bf4ff40b3843ffc2a47 finally found a fix (with lots of your help, thx). Please also read my description. I removed a test that I added before. If you think this test should be readded (in a version that works) please let me know.